crates/ruff_python_ast/src/visitor.rs

```diff
@@ -670,6 +697,22 @@ pub fn walk_with_item<'a, V: Visitor<'a> + ?Sized>(visitor: &mut V, with_item: &
     }
 }
 
+pub fn walk_type_param<'a, V: Visitor<'a> + ?Sized>(visitor: &mut V, type_param: &'a TypeParam) {
+    match type_param {
+        TypeParam::TypeVar(TypeParamTypeVar {
+            bound,
+            name: _,
+            range: _,
+        }) => {
+            if let Some(expr) = bound {
+                visitor.visit_expr(expr);
+            }
+        }
+        TypeParam::TypeVarTuple(_) => {}
+        TypeParam::ParamSpec(_) => {}
+    }
+}
```

Comment:
I don't know if any visitor overrides the function but we should call `visit_expr_context`:

```rust
Expr::Name(ast::ExprName { ctx, .. }) => {
	visitor.visit_expr_context(ctx);
}
```

Reply:
`name` here is an `Identifier` not a `Expr::Name` with context. Did you man somewhere else?

Reply:
Oh, my mistake. The fact that `name` is an identifier might be a bit misleading (this is unrelated to this PR).